packages/devtools_app/lib/src/shared/diagnostics/tree_builder.dart

```diff
@@ -249,44 +140,32 @@ Future<void> _addInstanceRefItems(
     }
   }
 
-  final result = await _getObject(
+  final result = await getObject(
     variable: variable,
     isolateRef: variable.ref!.isolateRef,
     value: instanceRef,
   );
+
   if (result is Instance) {
-    if (FeatureFlags.evalAndBrowse && ref?.heapSelection != null) {
-      final ref = variable.ref!;
-      variable.addChild(
-        DartObjectNode.references(
-          'references',
-          ObjectReferences(
-            refNodeType: RefNodeType.refRoot,
-            value: ref.value,
-            isolateRef: ref.isolateRef,
-            heapSelection: ref.heapSelection,
-          ),
-        ),
-        index: 0,
-      );
+    if (FeatureFlags.evalAndBrowse &&
+        ref != null &&
+        ref.heapSelection != null) {
```

Comment:
this can be simplified to ref?.heapSelection != null

Reply:
I need it to promote ref to non-nullable